### ISSUE TITLE
transfers.py improvements

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#5866]
- [x] file removal should allow for some to succeed, some to fail, report and requeue
- [ ] investigate --bulk option and --from_file options for file removal
- [ ] overall this is fragile against sudden termination and restart, need to make it less stateful e.g. using files and dropboxes to communicate between components, instead of relying on everything in memory
- [ ] conistently use fts3 easy python client, not direct calls to REST
- [ ] improve logs

possibly more...

**Comment by belforte** [comment]
and things should be coded such that all this code is common to rucio implementation so this effort is long lasting

**Comment by dciangot** [comment]
> * [ ]  file removal should allow for some to succeed, some to fail, report and requeue

still I see this as a lot of work that would not be in the right place in any case. IMO there is no reason for the delete process to be on the transfer tool. It should be in either the PJ (e.g, checking all is ok reporting status and only after remove the file) or in a central place after a timeout. 

In particular I disagree with the principle for which the file in tmp area has to be removed almost immediately. The moment you do that you are wasting a lot of CPU if something went weird in the transfer.

> * [ ]  investigate --bulk option and --from_file options for file removal

no need if put in the PJ, while needed for a central approach

> * [ ]  overall this is fragile against sudden termination and restart, need to make it less stateful e.g. using files and dropboxes to communicate between components, instead of relying on everything in memory

a part from the removal, everything should be already done in this way (cacthing errors and not updating the to-do list in case there is one). We do have a case in which this is not working though, but AFAIK we have no clue when yet. 

> * [ ]  conistently use fts3 easy python client, not direct calls to REST
> * [ ]  improve logs
> 
> possibly more...

to me, one important thing to be discussed (maybe again) is:

*[ ] make CRAB client able to retry transfers (of course only in the case we do not remove files from tmp)

I think that in many cases this could be what users are asking for, more true when we are in a CPU-limited period.

**Comment by belforte** [comment]
thanks for comments Diego. This list was indeed mostly as a reminder of things to talk about. I agree with some, but not all of what you wrote. Let's talk about it when you are back.
IIRC we discussed retries at length in the past and the question always is: what do PJ or user do that FTS can not do when it comes to trying again ? OTOH we ca surely live with some file removal failing and leaving to site cron to cleanup.
Surely we can look into delaying the temp. file deletion in case the user needs to rescue it, but... it may be difficult to define properly, anyhow...

**Comment by belforte** [comment]
A problem was identified and solved with file removal in temp area.
The other things are of the 'cleanup kind'.
So I am marking as non urgent.

**Comment by dciangot** [comment]
- [ ] make use of dictionaries instead of complicated lists
